pkg/helm/charts.go

```diff
@@ -95,12 +95,38 @@ func (c Charts) Vendor() error {
 	for _, r := range c.Manifest.Requires {
 		chartName := parseReqName(r.Chart)
 		chartPath := filepath.Join(dir, chartName)
-		if _, err := os.Stat(chartPath); err == nil {
-			log.Printf(" %s@%s exists", r.Chart, r.Version.String())
-			continue
+
+		_, err := os.Stat(chartPath)
+		if err == nil {
+			chartManifestPath := filepath.Join(chartPath, "Chart.yaml")
+			chartManifestBytes, err := ioutil.ReadFile(chartManifestPath)
+			if err != nil {
+				return fmt.Errorf("error reading chart manifest: %s", err)
+			}
+			var chartManifest map[string]interface{}
+			if err := yaml.Unmarshal(chartManifestBytes, &chartManifest); err != nil {
+				return fmt.Errorf("error unmarshalling chart manifest: %s", err)
+			}
+			currentChartVersion, ok := chartManifest["version"]
+			if !ok {
+				return fmt.Errorf("chart manifest %s does not contain a version field", chartManifestPath)
+			}
+			currentChartVersionStr := fmt.Sprintf("%v", currentChartVersion)
+
+			if currentChartVersionStr == r.Version.String() {
+				log.Printf(" %s@%s exists", r.Chart, r.Version.String())
+				continue
+			} else {
+				log.Printf("Removing %s@%s", r.Chart, r.Version.String())
+				if err := os.RemoveAll(chartPath); err != nil {
```

Comment:
RemoveAll is a bit scary. I'm reasonably happy that this code shouldn't delete anyone's entire project checkout (or home dir) unless they configure their chartfile directory to ".", or "../..", and that directory happens to contain a Chart.yaml. Perhaps it's still worth including a more stringent check? What do you reckon?